File: leetcode/3.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        char_set = set()

        left = 0
        max_lenght = 0

        for right in range(len(s)):
            while s[right] in char_set:
                char_set.remove(s[left])
                left += 1

            char_set.add(s[right])

            max_lenght = max(max_lenght, right - left + 1)

        return max_lenght 

File: leetcode/3.cpp
class Solution {
public:
  int lengthOfLongestSubstring(string s) {
    int ans = 0;
    vector<int> count(128);

    for (int left = 0, right = 0; right < s.length(); ++right) {
      ++count[s[right]];
      while (count[s[right]] > 1)
        --count[s[left++]];
      ans = max(ans, right - left + 1);
    }
    return ans;
  }
};